src/manifolds/SymplecticStiefel.jl

```diff
@@ -0,0 +1,664 @@
+    > Bendokat, Thomas and Zimmermann, Ralf:
+	> The real symplectic Stiefel and Grassmann manifolds: metrics, geodesics and applications
+	> arXiv preprint arXiv:2108.12447, 2021 (https://arxiv.org/abs/2108.12447)
+"""
+inverse_retract(::SymplecticStiefel, p, q, ::CayleyInverseRetraction)
+
+function inverse_retract!(M::SymplecticStiefel, X, p, q, ::CayleyInverseRetraction)
+    U_inv = lu!(add_scaled_I!(symplectic_inverse_times(M, p, q), 1))
+    V_inv = lu!(add_scaled_I!(symplectic_inverse_times(M, q, p), 1))
+
+    X .= 2 .* ((p / V_inv .- p / U_inv) .+ ((p + q) / U_inv) .- p)
+    return X
+end
+
+@doc raw"""
+    manifold_dimension(::Symplectic{n})
+
+Returns the dimension of the symplectic Stiefel manifold embedded in ``ℝ^{2n \times 2k}``,
+i.e. [^Bendokat2021]
+````math
+    \operatorname{dim}(\operatorname{SpSt}(2n, 2k)) = (4n - 2k + 1)k.
+````
+"""
+manifold_dimension(::SymplecticStiefel{n,k}) where {n,k} = (4n - 2k + 1) * k
+
+@doc raw"""
+    project(::Union{SymplecticStiefel,Symplectic}, p, A)
+    project!(::Union{SymplecticStiefel, Symplectic}, Y, p, A)
+
+Given a point ``p \in \operatorname{SpSt}(2n, 2k)``,
+project an element ``A \in \mathbb{R}^{2n \times 2k}`` onto
+the tangent space ``T_p\operatorname{SpSt}(2n, 2k)`` relative to
+the euclidean metric of the embedding ``\mathbb{R}^{2n \times 2k}``.
+
+That is, we find the element ``X \in T_p\operatorname{SpSt}(2n, 2k)``
+which solves the constrained optimization problem
+
+````math
+    \operatorname{min}_{X \in \mathbb{R}^{2n \times 2k}} \frac{1}{2}||X - A||^2, \quad
+    \text{s.t.}\;
+    h(X) = X^T Q p + p^T Q X = 0,
+````
+where ``h : \mathbb{R}^{2n \times 2k} \rightarrow \operatorname{skew}(2k)`` defines
+the restriction of ``X`` onto the tangent space ``T_p\operatorname{SpSt}(2n, 2k)``.
+"""
+project(::Union{SymplecticStiefel,Symplectic}, p, A)
+
+function project!(::Union{SymplecticStiefel,Symplectic}, Y, p, A)
+    Q = SymplecticMatrix(Y, p, A)
+    Q_p = Q * p
+
+    function h(X)
+        XT_Q_p = X' * Q_p
+        return XT_Q_p .- XT_Q_p'
+    end
+
+    # Solve for Λ (Lagrange mutliplier):
+    pT_p = p' * p  # (2k × 2k)
+    Λ = sylvester(pT_p, pT_p, h(A) ./ 2)
+
+    Y[:, :] = A .- Q_p * (Λ .- Λ')
+    return Y
+end
+
+@doc raw"""
+    rand(M::SymplecticStiefel{n, k})
+
+Generate a random point ``p \in \operatorname{SpSt}(2n, 2k)``
+by first generating a random symplectic matrix
+``p_{\operatorname{Sp}} \in \operatorname{Sp}(2n)``,
+and then projecting onto the Symplectic Stiefel manifold using the
+[`canonical_projection`](@ref).
+That is, ``p = π(p_{\operatorname{Sp}})``.
+"""
+function Base.rand(M::SymplecticStiefel{n,k}; hamiltonian_norm=1 / 2) where {n,k}
+    p_symplectic = rand(Symplectic(2n); hamiltonian_norm=hamiltonian_norm)
+    return canonical_projection(M, p_symplectic)
+end
+
+@doc raw"""
+    random_vector(::SymplecticStiefel{n, k}, p)
+
+The symplectic Stiefel tangent space at ``p`` can be parametrized as [^Bendokat2021]
+````math
```

Comment:
I think it would be better do document the tangent space in the SymplecticStiefel type and here just refer to `[`SymplecticStiefel`](@ref)` - that way the form is documented in a central place

If you see a good way to split this focal into multiple lines, that would be great, its quite long).

Reply:
I've moved the documentation into the `SymplecticStiefel` type now, and removed the documentation of
the `random_vector` function entirely in favor of documenting the new `rand(::SymplecticStiefel; vector_at=...)` function 
more thoroughly.